Microsoft.NET.Build.Containers/AuthHandshakeMessageHandler.cs

```diff
@@ -107,7 +109,9 @@ private async Task<string> GetTokenAsync(Uri realm, string service, string scope
         }
 
         // save the retrieved token in the cache
-        TokenCache[realm.Host] = token.ResolvedToken;
+        lock(TokenCacheInsertLock) {
```

Comment:
this indexer was the only contended part of the operation that I could see.  we could lock, or we could do a `TryAdd`.

A more sophisticated/correct version of this would actually lock at the token-retrieval operation on a per-host level, now that I'm thinking about it.

Reply:
Switch to a `ConcurrentDictionary`, to use `AddOrUpdate` and get that finer-grained locking for free?

Reply:
done!